ts/smelter/src/components/Mp4.ts

```diff
@@ -32,17 +32,23 @@ function Mp4(props: Mp4Props) {
   useEffect(() => {
     const newInputId = newInternalStreamId();
     setInputId(newInputId);
-    const pathOrUrl: Pick<RegisterMp4Input, 'url' | 'serverPath'> =
-      props.source.startsWith('http://') || props.source.startsWith('https://')
-        ? { url: props.source }
-        : { serverPath: props.source };
+
+    let source: Pick<RegisterMp4Input, 'url' | 'serverPath' | 'blob'>;
```

Comment:
source might be confused with propsSource, maybe rename it to sourceObject or something like this

Also unrelated to your changes, but modify
```
const { muted, volume, ...otherProps } = props;
```
to 
```
const { muted, volume, source, ...otherProps } = props;
```

because we pass otherProps to InnerInputStream component, and source prop should not be passed there